scripts/zones/Grand_Palace_of_HuXzoi/mobs/Jailer_of_Temperance.lua

```diff
@@ -21,26 +47,37 @@ entity.onMobSpawn = function(mob)
     mob:setMod(xi.mod.PIERCE_SDT, 0)
     mob:setMod(xi.mod.IMPACT_SDT, 1000)
     -- Set the magic resists. It always takes no damage from direct magic
-    for n =1, #xi.magic.resistMod, 1 do
-        mob:setMod(xi.magic.resistMod[n], 0)
-    end
-    for n =1, #xi.magic.specificDmgTakenMod, 1 do
-        mob:setMod(xi.magic.specificDmgTakenMod[n], 10000)
-    end
+    mob:setMod(xi.mod.DMGMAGIC, -99)
```

Comment:
99% ?

I'm tired so maybe I am missing something but I thought you wanted 100% for no damage per the comment. I vaguely recall a reason in the past for someone using the element mods instead of straight magic damage reduction but I'd need to test a non elemental spell on retail to know for sure, and its their fault for not explaining if they really needed it that way - so it can be like you got it just make sure the value is what you needed.